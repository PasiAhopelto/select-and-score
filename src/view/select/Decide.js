import React, { Component } from 'react';

export const Decision = { YES: 1, NO: 2, UNDECIDED: 3 }

export default class Decide extends Component {
  constructor(props) {
    super(props);
    this.state = { decision: Decision.UNDECIDED };
  }

  render() {
    return <div decision={this.state.decision}></div>;
  }
}
