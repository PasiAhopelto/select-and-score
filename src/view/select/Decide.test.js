import React from 'react';
import Decide from './Decide'
import { Decision } from './Decide'
import renderer from 'react-test-renderer';

it('Is initially UNDECIDED', () => {
  const component = renderer.create(
    <Decide />
  );
  let decision = component.toJSON();
  expect(decision).toMatchSnapshot();
  expect(decision.props.decision).toBe(Decision.UNDECIDED);
});
