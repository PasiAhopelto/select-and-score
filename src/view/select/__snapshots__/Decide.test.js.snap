// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is initially UNDECIDED 1`] = `
<div
  decision={3}
/>
`;
